matchms/filtering/remove_peaks_around_precursor_mz.py

```diff
@@ -0,0 +1,28 @@
+import numpy
+from ..Spikes import Spikes
+from ..typing import SpectrumType
+
+
+def remove_peaks_around_precursor_mz(spectrum: SpectrumType, mz_tolerance: float = 17) -> SpectrumType:
+
+    """Remove peaks that are within mz_tolerance (in Da) of
+       the precursor mz, exlcuding the precursor peak.
+
+    Parameters
+    ----------
+    spectrum
+        Input spectrum.
+    mz_tolerance
+        Tolerance of mz values that are not allowed to lie
+        within the precursor mz. Default is 17 Da.
+    """
+
+    assert mz_tolerance >= 0, "mz_tolerance must be a positive floating point."
+    precursor_mz = spectrum.get("precursor_mz")
+    mzs, intensities = spectrum.peaks
+    if precursor_mz:
```

Comment:
If there is no precursor then spectrum is returned. I expected an exception to be thrown because the filter requires a precursor. See for example [add_losses()](https://github.com/matchms/matchms/blob/master/matchms/filtering/add_losses.py#L25)

Reply:
Just saw that also in ``add_losses()`` we have the assert after ``if precursor_mz``. 
We can 1) add the same assert function here. In that case we accept missing precursor_mz entries, but if there is an entry it must be a number.
What you suggest here is 2) add the assert before the if and make the function fail when there is no numerical ``precursor_mz``. To me that's fine as well and indeed seems cleaner. We just would have to check if that breaks something.

Reply:
2) would be clearer to me also. Can't really see it breaking anything in the filter function or add_losses.

Reply:
Let's do 2 then.

Reply:
I think we should also add the same thing for `require_precursor_below_mz` for consistency.

Reply:
Definitely! Good point.

Reply:
Some of the tests fail when we add an assertion for the missing precursor mz. For `require_precursor_below_mz` the cloning test fails (raises an assertion error for missing precursor mz) and the test for when the precursor mz is absent fails as well. I suppose we can just remove the test missing precursor mz and add a dummy precursor mz to the cloning test?

Reply:
Yes, we can simply add a precursor_mz to the ``test_require_precursor_below_mz``.

Reply:
Ok done. Shall I also make the same changes for add_losses?

Reply:
I'change add_losses in a separate, tiny PR, just to not make this PR even bigger. (I also added an issue for that).

Reply:
Ok great.